k9mail/src/main/java/com/fsck/k9/mailstore/ProvidedTempFileBody.java

```diff
@@ -82,11 +88,11 @@ public long getSize() {
         throw new IllegalStateException("Data must be fully written before it can be read!");
     }
 
-    public File getFile() throws IOException {
+    public Uri getProviderUri(String mimeType) throws IOException {
```

Comment:
But that way, `AttachmentInfoExtractor` will have to special-case the file it receives from the body to work with the fileprovider out of nowhere, or am I missing a better way to do this?

The FileProvider also does not know anything about bodies at the moment..